Gems/Prefab/PrefabBuilder/PrefabGroup.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "PrefabGroup.h"
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/RTTI/BehaviorContext.h>
+#include <AzCore/JSON/error/error.h>
+#include <AzCore/JSON/error/en.h>
+
+namespace AZ::SceneAPI::SceneData
+{
+     PrefabGroup::PrefabGroup()
+        : m_id(Uuid::CreateNull())
+        , m_name()
+    {
+         m_prefabDom.SetNull();
+    }
+
+    const AZStd::string& PrefabGroup::GetName() const
+    {
+        return m_name;
+    }
+
+    void PrefabGroup::SetName(AZStd::string&& name)
+    {
+        m_name = AZStd::move(name);
+    }
+
+    const Uuid& PrefabGroup::GetId() const
+    {
+        return m_id;
+    }
+
+    void PrefabGroup::SetId(Uuid&& id)
+    {
+        m_id = AZStd::move(id);
+    }
+
+    Containers::RuleContainer& PrefabGroup::GetRuleContainer()
+    {
+        return m_rules;
+    }
+
+    const Containers::RuleContainer& PrefabGroup::GetRuleContainerConst() const
+    {
+        return m_rules;
+    }
+            
+    DataTypes::ISceneNodeSelectionList& PrefabGroup::GetSceneNodeSelectionList()
+    {
+        return m_nodeSelectionList;
+    }
+
+    const DataTypes::ISceneNodeSelectionList& PrefabGroup::GetSceneNodeSelectionList() const
+    {
+        return m_nodeSelectionList;
+    }
+
+    void PrefabGroup::SetPrefabDom(AzToolsFramework::Prefab::PrefabDom&& prefabDom)
+    {
+        m_prefabDom = AZStd::move(prefabDom);
+        m_prefabDomBuffer.clear();
+    }
+
+    void PrefabGroup::SetPrefabDomBuffer(AZStd::string&& prefabDomBuffer)
```

Comment:
Why not declare the argument by value, so `AZStd::string prefabDomBuffer` so this function can be called with an l-value and r-value instead of only with r-values?